### PR TITLE
[BE-06] Add rating 

### PR DESCRIPTION
**Pull request by PawelKrol23**
https://trello.com/c/c0qcR71t

### CODE REVIEW

File: app/controllers/api/v1/rates_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+module Api
+  module V1
+    class RatesController < ApplicationController
+      before_action :set_recipe
+      def create
```
Comment (grabomarcin) [inline app/controllers/api/v1/rates_controller.rb]:
```suggestion
      before_action :set_recipe
      
      def create
```

File: app/serializers/recipe_show_serializer.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 class RecipeShowSerializer < RecipeSerializer
   NIL_RESULT = 'unknown'.freeze
 
-  attributes :cooking_time, :video_link, :level
-  
+  attributes :level, :video_link, :cooking_time
+  has_many :rates
```
Comment (grabomarcin) [inline app/serializers/recipe_show_serializer.rb]: I think that you should return `avg_rate`.  Returning collection of votes is pointless. You can fix that in a fast way.

File: db/schema.rb [hunk 3]
```diff
@@ -24,15 +24,22 @@
...
+  end
+
   create_table "recipes", force: :cascade do |t|
     t.string "name", null: false
     t.text "content"
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
     t.string "video_link"
-    t.integer "cooking_time"
-    t.decimal "price"
```
Comment (grabomarcin) [inline db/schema.rb:34]: Why price is removed?

File: spec/controllers/api/v1/rates_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    context 'with valid votes' do
+      it 'creates new rate' do
+        expect do
+          post :create, params: { recipe_id: recipe.id, rate: rate_params }
+        end.to change(Recipe, :count).by(1)
+      end
+    end
+
+    context 'with invalid votes' do
+      let!(:recipe) { create(:recipe) }
```
Comment (grabomarcin) [inline spec/controllers/api/v1/rates_controller_spec.rb]: Why do you create `recipe` for the second time? Can't you just use recipe created in the first `let`?

File: spec/factories/rate.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
```
Comment (grabomarcin) [inline spec/factories/rate.rb]: The file should be named `rates.rb`

File: spec/models/rate_spec.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe Rate, type: :model do
+  describe 'associations' do
+    it { is_expected.to belong_to(:recipe) }
+  end
+
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:vote) }
+    it { is_expected.to validate_inclusion_of(:vote).in_range(1..5).with_message(described_class::INCLUSION_ERROR_MESSAGE) }
```
Comment (grabomarcin) [inline spec/models/rate_spec.rb]: This line is probably too long, you can run rubocop.

File: app/controllers/api/v1/rates_controller.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        rate = Rate.new(rate_params)
+        rate.recipe = @recipe
+        if rate.save
```
Comment (DanielOldak) [inline app/controllers/api/v1/rates_controller.rb:8]: It's good pattern to use a "parent" object as a foundation to create an association.
In this case you can go with ```if @recipe.rates.create(rate_params)```

File: app/serializers/recipe_show_serializer.rb (hunk 2 above)
Comment (DanielOldak) [reply app/serializers/recipe_show_serializer.rb]: Good point Marcin 👌 having avg_rate and maybe rates_count makes more sense

File: db/schema.rb (hunk 3 above)
Comment (PawelKrol23) [reply db/schema.rb:34]: It shouldn't be there. Price field is still in PR and it isn't in any migration.

File: app/serializers/recipe_show_serializer.rb [hunk 8]
```diff
@@ -26,4 +25,13 @@ def video_link
 
     object.video_link
   end
+
+  def avg_rate
+    # returns 0 if there is no rates
+    object.rates.average(:vote) || 0
```
Comment (plokijro) [inline app/serializers/recipe_show_serializer.rb:31]: 👌

File: spec/controllers/api/v1/rates_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+describe Api::V1::RatesController do
+  describe '[GET] #create' do
```
Comment (plokijro) [inline spec/controllers/api/v1/rates_controller_spec.rb]:
```suggestion
  describe '[POST] #create' do
```

**Review by grabomarcin — APPROVED** [review]

**Pull request merged**